pkgs/development/python-modules/cupy/default.nix

```diff
@@ -17,7 +17,7 @@
 }:
 
 let
-  inherit (cudaPackages) cudnn cutensor nccl;
+  inherit (cudaPackages) cudnn_8_9 cutensor nccl;
```

Comment:
I'd say the "officially supported" way to do this is to spawn a new `cudaPackages = prev.cudaPackages.overrideScope (cu-fin: _: { cudnn = cu-fin.cudnn_8_9; })` and pass that to `cupy`, but we can try and keep this as is for now